File: src/seven.py
#!/usr/bin/env python


def main():
  in_str = input()
  in_nums = list(map(int, in_str.split(',')))
  max_num = max(in_nums)

  min_so_far = 1000007
  for i in range(max_num):
    s = sum([abs(x - i) for x in in_nums])
    if s < min_so_far:
      min_so_far = s

  print(min_so_far)

  min_so_far = 100000007
  for i in range(max_num):
    s = sum([abs(x - i) * (abs(x - i) + 1) // 2 for x in in_nums])
    if s < min_so_far:
      min_so_far = s

  print(min_so_far)


if __name__ == "__main__":
  main()

File: src/seven.rs
fn main() {
  let in_str: &str = "16,1,2,0,4,2,7,1,2,14";

  // part one
  let v: Vec<i32> = in_str.split(',').map(|x| x.parse::<i32>().unwrap()).collect();
  let v_max = v.iter().max().unwrap();
  let mut min_so_far: i32 = 1000007;
  for i in 1..*v_max {
    let mut count = 0;
    for vs in &v {
      count += (vs - i).abs();
    }
    if count < min_so_far {
      min_so_far = count;
    }
  }
  println!("part one: {:?}", min_so_far);

  // part two
  let mut min_so_far: i32 = 100000007;
  for i in 1..*v_max {
    let mut count = 0;
    for vs in &v {
      count += (vs - i).abs() * ((vs - i).abs() + 1) / 2;
    }
    if count < min_so_far {
      min_so_far = count;
    }
  }
  println!("part two: {:?}", min_so_far);
}
